openwisp_monitoring/monitoring/base/models.py

```diff
@@ -151,9 +171,12 @@ def _get_or_create(cls, **kwargs):
             if lookup_kwargs.get('name'):
                 del lookup_kwargs['name']
             extra_tags = lookup_kwargs.pop('extra_tags', {})
-            metric = cls.objects.get(**lookup_kwargs)
+            metric_cache_key = get_or_create_metric_cache_key(**kwargs)
```

Comment:
let's use another private method to fetch the metric object and add the memoize decorator on it